### PR TITLE
Check classes and functions documented for api module

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Findings.kt [hunk 1]
```diff
@@ -44,13 +47,23 @@ interface HasEntity {
...
  * Provides a compact string representation.
  */
 interface Compactable {
+    /**
+     * Contract to format implementing object to a string representation.
+     */
     fun compact(): String
+
+    /**
+     * Same as [compact] except the content should contain a substring which
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Findings.kt]: Something is missing here.

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/MultiRule.kt [hunk 2]
```diff
@@ -19,6 +24,10 @@ abstract class MultiRule : BaseRule() {
         }
     }
 
+    /**
+     * Preferred way to run child rules because this composite rule
+     * takes care of evaluating if a specific child run should be run at all.
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/MultiRule.kt]: Extra run

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssign.kt [hunk 3]
```diff
@@ -10,11 +10,17 @@ class SingleAssign<T : Any> {
 
     private lateinit var _value: T
 
+    /**
+     * Gets the stored value if it is initialized or else an error is thrown.
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssign.kt]: Weird sentence

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssign.kt [hunk 4]
```diff
@@ -10,11 +10,17 @@ class SingleAssign<T : Any> {
...
+     */
     operator fun getValue(thisRef: Any?, property: KProperty<*>): T {
         check(this::_value.isInitialized) { "Property ${property.name} has not been assigned yet!" }
         return _value
     }
 
+    /**
+     * Only sets the value if it was not yet set or else an error is thrown
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssign.kt]: Again weird sentence

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt [hunk 5]
```diff
@@ -26,10 +26,42 @@ class SplitPattern(
         then: (Sequence<T>) -> Sequence<T>
     ): Sequence<T> = if (condition) then(this) else this
 
+    /**
+     * Does any part contain given [value]?
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt]: A given value

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt [hunk 6]
```diff
@@ -26,10 +26,42 @@ class SplitPattern(
...
+    fun any(value: String?): Boolean = excludes.any { value?.equals(it, ignoreCase = true) == true }
+
+    /**
+     * Tests if none of the parts contain the given [value].
+     */
     fun none(value: String): Boolean = !contains(value)
+
+    /**
+     * Finds all parts which match given [value].
+     */
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt]: The

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt [hunk 7]
```diff
@@ -26,10 +26,42 @@ class SplitPattern(
...
+    @Deprecated(
+        "The name 'equals' should only be used when matching the 'equals contract'.",
+        replaceWith = ReplaceWith("any(value)")
+    )
     fun equals(value: String?): Boolean = excludes.any { value?.equals(it, ignoreCase = true) == true }
+
+    /**
+     * Is there any element which matches given [value]?
+     */
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt]: The

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt [hunk 8]
```diff
@@ -26,10 +26,42 @@ class SplitPattern(
...
+
+    /**
+     * Finds all parts which match given [value].
+     */
     fun matches(value: String): List<String> = excludes.filter { value.contains(it, ignoreCase = true) }
-    fun startWith(name: String?): Boolean = excludes.any { name?.startsWith(it) ?: false }
+
+    /**
+     * Tests if any part starts with given [value]
+     */
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt]: The

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt [hunk 9]
```diff
@@ -26,10 +26,42 @@ class SplitPattern(
...
+    fun startWith(value: String?): Boolean = excludes.any { value?.startsWith(it) ?: false }
+
+    /**
+     * Transforms all parts by given [transform] function.
+     */
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SplitPattern.kt]: The

**Comment by arturbosch** [comment]
Thanks for grammar checking!

**Pull request merged**